src/daft-dsl/src/python_udf.rs

```diff
@@ -0,0 +1,177 @@
+use std::{fmt::Display, sync::Arc};
+
+use common_error::DaftResult;
+use daft_core::{
+    prelude::{DataType, Field, Schema},
+    series::Series,
+};
+use itertools::Itertools;
+use serde::{Deserialize, Serialize};
+
+use crate::{
+    functions::{python::RuntimePyObject, scalar::ScalarFn},
+    Expr, ExprRef, LiteralValue,
+};
+
+#[derive(derive_more::Display, Debug, Clone, Serialize, Deserialize, PartialEq, Eq, Hash)]
+#[display("{_0}")]
+pub enum PyScalarFn {
+    RowWise(RowWisePyFn),
+}
+
+impl PyScalarFn {
+    pub fn call(&self, args: Vec<Series>) -> DaftResult<Series> {
+        match self {
+            Self::RowWise(func) => func.call(args),
+        }
+    }
+
+    pub fn children(&self) -> Vec<ExprRef> {
+        match self {
+            Self::RowWise(RowWisePyFn { args, .. }) => args.clone(),
+        }
+    }
+
+    pub fn with_new_children(&self, children: Vec<ExprRef>) -> Self {
+        match self {
+            Self::RowWise(func) => Self::RowWise(func.with_new_children(children)),
+        }
+    }
```

Comment:
nit: inputs. The inputs are children in the expression tree, but this type is not related to the expression tree. The expression tree is built up from this type.